### DISCUSSION TITLE
Support ObservableInput consistently as the return type for selector fns (like delayWhen)

### DISCUSSION THREAD

**Discussion by jollytoad** [ReactiveX/rxjs#6972] [Ideas / Feature request]
Some operators such as `bufferWhen`, `windowWhen`, `audit` allow their selector fn to return an `ObservableInput` which really helps with interop with native JS features, such as Promises, Iterables/AsyncIterables etc.

But the `delayWhen` delayDurationSelector fn must return an `Observable`, it would really help if this was also `ObservableInput`. I'm not sure if this is intentional (I don't understand why it would be), or an oversight. I'm not sure if this is the case elsewhere, but it would certainly help if it was consistent across all selector fns.

**Comment by benlesh** [comment]
Core Team: `delayWhen` should definitely be fixed. We'll need to look to see what other operators have this issue.

**Reply by jakovljevic-mladen** [reply]
Created issue for `delayWhen`: #7046

**Reply by jakovljevic-mladen** [reply]
So far, I've found that:

- `buffer`'s `closingNotifier`,
- `distinct`'s `flushes`,
- `repeatWhen`'s `notifier`,
- `sample`'s `notifier`,
- `retryWhen`'s `notifier`,
- `sequenceEqual`'s `compareTo`,
- `ShareConfig`'s `resetOnError`, `resetOnComplete` and `resetOnRefCountZero`,
- `skipUntil`'s `notifier` and
- `window`'s `windowBoundaries`

could potentially extend `Observable` to `ObservableInput`. @benlesh, could you please review the list and let me know which of these operators should be updated?

**Reply by benlesh** [reply]
Yes, these should all be `ObservableInput`.
